apps/www/_blog/2023-09-05-pgvector-v0-5-0-hnsw.mdx

```diff
@@ -0,0 +1,264 @@
+    className="hidden dark:block"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-ivfflat-vs-hnsw--dark.png"
+  />
+</div>
+
+When we maintain fixed HNSW build parameters, we can adjust the select query parameter `ef_search` to balance query speed and precision. To achieve precision@10 of 0.98, we increased it from the default 40 to 100. For precision@10 of 0.99, we further raised it to 250. Remarkably, HNSW demonstrated over six times better performance while maintaining the same level of precision. With higher precision@10 of 0.99 HNSW even outperforms [qdrant](https://nirantk.com/writing/pgvector-vs-qdrant/) on equivalent compute resources.
+
+<div>
+  <img
+    alt="dbpedia embeddings comparing qdrant and hnsw queries-per-second (light)"
+    className="dark:hidden"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-qdrant-vs-hnsw--light.png"
+  />
+  <img
+    alt="dbpedia embeddings comparing qdrant and hnsw queries-per-second (dark)"
+    className="hidden dark:block"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-qdrant-vs-hnsw--dark.png"
+  />
+</div>
+
+### Scaling the database
+
+Performance scales predictably with the size of the database for the HNSW index, just as [it does for the IVFFlat index](https://supabase.com/blog/pgvector-performance#scaling-the-database). The difference in performance between IVFFlat and HNSW remains consistent after a compute upgrade.
+
+<div>
+  <img
+    alt="dbpedia embeddings comparing ivfflat and hnsw queries-per-second using the 4XL compute addon (light)"
+    className="dark:hidden"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-ivfflat-vs-hnsw-4xl--light.png"
+  />
+  <img
+    alt="dbpedia embeddings comparing ivfflat and hnsw queries-per-second using the 4XL compute addon (dark)"
+    className="hidden dark:block"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-ivfflat-vs-hnsw-4xl--dark.png"
+  />
+</div>
+
+Switching to a 4XL compute add-on (with a 16-core CPU and 64GB of RAM) resulted in a 69% increase in QPS for a precision@10 of 0.99 compared to 2XL instance.
+
+<div>
+  <img
+    alt="dbpedia embeddings comparing hnsw queries-per-second using the 2XL vs 4XL compute addon (light)"
+    className="dark:hidden"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-hnsw-2xl-vs-4xl--light.png"
+  />
+  <img
+    alt="dbpedia embeddings comparing hnsw queries-per-second using the 2XL vs 4XL compute addon (dark)"
+    className="hidden dark:block"
+    src="/images/blog/2023-09-05-pgvector-v0-5-0-hnsw/dbpedia-hnsw-2xl-vs-4xl--dark.png"
+  />
+</div>
+
+Furthermore, having 64GB of RAM is better suited for this type of workload because Postgres uses approximately 30-35GB of RAM to achieve optimal performance. With a 4XL setup, you not only have the capacity to store 1,000,000 vectors but also the flexibility to accommodate additional data or increase the number of vectors as needed.
```

Comment:
```suggestion
Furthermore, having 64GB of RAM is better suited for this dataset because Postgres uses approximately 30-35GB of RAM to achieve optimal performance. With a 4XL setup, you not only have the capacity to store 1,000,000 vectors but also the flexibility to accommodate additional data or increase the number of vectors as needed.
```